students/JaeKim/session06/mailroom.py

```diff
@@ -0,0 +1,127 @@
+import tempfile
+
+#!/usr/bin/env python3
+
+"""
+mailroom assignment
+"""
+
+players = {
+    "LeBron James": [1000000, 2000000, 3000000],
+    "Dwyane Wade": [100000, 20000, 30000],
+    "Carmelo Anthony": [1],
+    "Kevin Durant": [1, 2],
+    "Michael Jordan": [2]
+}
+
+
+def reset_donors():
+    return {
+        "LeBron James": [1000000, 2000000, 3000000],
+        "Dwyane Wade": [100000, 20000, 30000],
+        "Carmelo Anthony": [1],
+        "Kevin Durant": [1, 2],
+        "Michael Jordan": [2]
+    }
+
+
+def safe_input(message=""):
+    try:
+        get_input = input ("=> " + message)
+    except KeyboardInterrupt:
+        return None
+    except EOFError:
+        return None
+    return get_input
+
+
+def send_thank_you(name):
+    donation = sum(players[name])
+
+    email = ("Hi {},\n\n"
+             "Thank you for your donation of {} to the research center.\n\n"
+             "Have a great day!".format(name, donation))
+
+    return email
+
+
+def add_donor(donor_name, donor_amount):
+    players[donor_name] = [int(donor_amount)]
+    print("Hi {},\n\n"
+          "Thank you for your donation of {} to the research center.\n\n"
+          "Have a great day!".format(donor_name, donor_amount))
+
+
+def thank_you(name):
+    if name.lower() == "list":
+        for player in players:
+            print(player, players[player])
+
+        print("\n")
+    elif name not in players:
+        print("Enter a donation for " + name)
+        donation = safe_input()
+        if donation is None:
+            return
+        add_donor(name, donation)
+    elif name in players:
+        send_thank_you(name)
+
+
+def send_all_donors():
+    for player in players:
+        donation_sum = sum(players[player])
+        email_path = tempfile.gettempdir() + '\\' + player + ".txt"
```

Comment:
This is an issue for a non-Windows environment.  Recommend a solution that works for any platform.